pyvista/plotting/plotting.py

```diff
@@ -4585,6 +4594,31 @@ def export_obj(self, filename):
         exporter.SetRenderWindow(self.ren_win)
         return exporter.Write()
 
+    @property
+    def _datasets(self):
+        """List of all datasets associated with this plotter."""
+        datasets = []
+        for renderer in self.renderers:
+            for actor in renderer.actors.values():
+                mapper = actor.GetMapper()
+
+                # ignore any mappers whose inputs are not datasets
+                if hasattr(mapper, 'GetInputAsDataSet'):
+                    datasets.append(mapper.GetInputAsDataSet())
+
+        return datasets
+
+    def _remove_added_scalars(self):
+        """Remove any scalars added to the datasets."""
+        for mesh, (name, assoc) in self._added_scalars:
+            if name is None:
+                continue
```

Comment:
If we remove `None` cases before addition we can remove this branch.

Reply:
Resolved in 32fcae6f21b8213e6c988a2d78a960e9a0300d09.